src/porepy/models/slightly_compressible_flow_model.py

```diff
@@ -109,63 +95,27 @@ def _assign_discretizations(self) -> None:
         """
 
         super()._assign_discretizations()
-        gb = self.gb
-        grid_list = [g for g, _ in gb.nodes()]
-
-
-        # Ad representation of discretizations
-        mass_mat = pp.MassMatrix(self.parameter_key)
-        mass_mat.discretize(gb)
-
-        # Ad variables
-        p = self._eq_manager.merge_variables([(g, self.variable) for g in grid_list])
-
-        # Update equations
-        subdomain_flow_eq = self._eq_manager.equations["subdomain_flow"]
-        interface_flow_eq = self._eq_manager.equations["interface_flow"]
-
-        mass = (
-            mass_mat * (p - p.previous_timestep()) / self.time_step
+        
+        if not hasattr(self, "dof_manager"):
+            self.dof_manager = pp.DofManager(self.gb)
+        
+        # Collection of subdomains
+        subdomains: List[pp.Grid] = [g for g, _ in self.gb]
+        
+        # AD representation of the mass operator
+        accumulation_term = pp.ad.MassMatrixAd(self.parameter_key, subdomains)
+        
+        # Need to find a way / refactor to access the superclass "and pressure" instead of create a new one
+        # If for some reason the mesh ordering change, this will be inconsistent
```

Comment:
@keileg your solution in branch [multiphase_flow_reactive_base](https://github.com/pmgbergen/porepy/blob/multiphase_reactive_base/src/porepy/models/multiphase_reactive_transport.py) with the AdVariables dataclass might help. Can we generalize that class and include it in the abstract model ( @IvarStefansson ), so that every model has access to the MergedVariable instances (global variables) defined on the whole domain?

Reply:
The MergedVariable instances which are defined in e.g. a parent class method

Reply:
Something along that line is indeed the plan

Reply:
Instead of creating a dual storage class, we can use a dictionary to store AD variables. The key is the already defined string representation for the variable e.g. "self.variable". I have updated the branch with this idea.

Reply:
@vlipovac as @IvarStefansson  said, we'll get there, but the concept should be clearer before we promote it to the base class.

@OmarDuran I vote for a class in this case, since that makes documentation of attributes more natural. Simpler syntax is also a plus.

Reply:
Alright, I have created a dual storage class for AD variables.